lib/dependencies/AMDDefineDependency.js

```diff
@@ -2,113 +2,137 @@
 	MIT License http://www.opensource.org/licenses/mit-license.php
 	Author Tobias Koppers @sokra
 */
-var NullDependency = require("./NullDependency");
+"use strict";
+const NullDependency = require("./NullDependency");
 
-function AMDDefineDependency(range, arrayRange, functionRange, objectRange) {
-	NullDependency.call(this);
-	this.range = range;
-	this.arrayRange = arrayRange;
-	this.functionRange = functionRange;
-	this.objectRange = objectRange;
+class AMDDefineDependency extends NullDependency {
+	constructor(range, arrayRange, functionRange, objectRange) {
+		super();
+		this.range = range;
+		this.arrayRange = arrayRange;
+		this.functionRange = functionRange;
+		this.objectRange = objectRange;
+	}
+
+	get type() {
+		return "amd define";
+	}
 }
-module.exports = AMDDefineDependency;
 
-AMDDefineDependency.prototype = Object.create(NullDependency.prototype);
-AMDDefineDependency.prototype.constructor = AMDDefineDependency;
-AMDDefineDependency.prototype.type = "amd define";
+AMDDefineDependency.Template = class AMDDefineDependencyTemplate {
+	get definitions() {
+		return {
```

Comment:
The getter for definitions is in the prototype, though, while `static` puts it in the constructor. Not sure why you need it to be available on `this.definitions`; it's fine to close over a module-level `const definitions` :)